src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -294,72 +295,84 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
             ScriptFile[] referencedFiles,
             Workspace workspace)
         {
-            if (foundSymbol != null)
+            if (foundSymbol == null)
             {
-                int symbolOffset = referencedFiles[0].GetOffsetAtPosition(
-                    foundSymbol.ScriptRegion.StartLineNumber,
-                    foundSymbol.ScriptRegion.StartColumnNumber);
+                return null;
+            }
 
-                // Make sure aliases have been loaded
-                await GetAliases();
+            int symbolOffset = referencedFiles[0].GetOffsetAtPosition(
+                foundSymbol.ScriptRegion.StartLineNumber,
+                foundSymbol.ScriptRegion.StartColumnNumber);
 
-                // We want to look for references first in referenced files, hence we use ordered dictionary
-                var fileMap = new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
-                foreach (ScriptFile file in referencedFiles)
-                {
-                    fileMap.Add(file.FilePath, file);
-                }
+            // Make sure aliases have been loaded
+            await GetAliases();
 
-                var allFiles = workspace.EnumeratePSFiles();
-                foreach (var file in allFiles)
+            // We want to look for references first in referenced files, hence we use ordered dictionary
+            var fileMap = new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
+            foreach (ScriptFile file in referencedFiles)
+            {
+                fileMap.Add(file.FilePath, file);
+            }
+
+            IEnumerable<string> allFiles = workspace.EnumeratePSFiles();
+            foreach (string file in allFiles)
+            {
+                if (!fileMap.Contains(file))
                 {
-                    if (!fileMap.Contains(file))
+                    ScriptFile scriptFile;
+                    try
                     {
-                        fileMap.Add(file, new ScriptFile(file, null, this.powerShellContext.LocalPowerShellVersion.Version));
+                        scriptFile = new ScriptFile(
```

Comment:
Oh? This doesn't try to pull an existing `ScriptFile` instance?  That explains why find references code lens is slow...

Don't know if you want to try this in this PR, but if this could be replaced with `Workspace.GetFile()` it would speed up finding references *significantly*.  Right now it's only looking at the cache for files that are explicitly referenced (which is pretty rare)

Reply:
Yeah I was a bit skeptical of that constructor. I'll see if changing it over works